tuf/api/metadata.py

```diff
@@ -116,6 +116,33 @@ def __init__(self, signed: T, signatures: Dict[str, Signature]):
         self.signed: T = signed
         self.signatures = signatures
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Metadata):
+            return False
+
+        # Verifies that the signatures for both objects are the same.
+        # TODO: Use Signature.__eq__ (secure-systems-lab/securesystemslib#383)
+        # pylint: disable=unidiomatic-typecheck
+        if not (type(self.signatures) == type(other.signatures) == dict):
+            return False
+
+        # Verifies that signatures keyids for both objects are the same
```

Comment:
this comment explains only the obvious -- "why" is what requires explanation (at least I assume the reason is to ensure the dict ordering as dict does not take care of it)